### PR TITLE
feat(api/v2): implement U2F session check

### PR DESCRIPTION
**Pull request by muhlemmer**
This change adds U2F support to the new (v2alpha) Session API. As the WebAuthN mechanism is the same as passkeys, the API is refactored to accommodate both U2F and Passkey use-cases with the same API Request message format.

The requested challenges are now message type, instead of a repeated enum. `RequestChallenges` can be used to obtain a `WebAuthN` challenge, with the needed parameters such as `domain` and `user_verification_requirement`.

- `domain` was moved out of the `CreateSessionRequest` as it is only needed for `WebAuthN`
- `user_verification_requirement` controls if the WebAuthN is a passkey or U2F level verification.

Updates #5477

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by muhlemmer** [comment]
In principle ready to review, I just have one small TODO to adjust the passkey error messages into webAuthN:  https://github.com/zitadel/zitadel/blob/4b8c0c4d7196d84be17ff688d3d646ae2ff067c3/internal/command/session_passkey.go#L70, but I have some issues with the asian character sets on my laptop. I'll get onto that tomorrow.

File: internal/query/sessions_test.go [hunk 1]
```diff
@@ -81,6 +80,7 @@ var (
 		"password_checked_at",
 		"intent_checked_at",
 		"passkey_checked_at",
```
Comment (stebenz) [inline internal/query/sessions_test.go:82]: should this not be webauthn_checked_at?

File: internal/query/sessions_test.go [hunk 2]
```diff
@@ -81,6 +80,7 @@ var (
...
+		"u2f_checked_at",
```
Comment (stebenz) [inline internal/query/sessions_test.go:83]: Should this not be webauth_user_verified?

File: internal/query/sessions_test.go [hunk 3]
```diff
@@ -102,6 +101,7 @@ var (
 		"password_checked_at",
 		"intent_checked_at",
 		"passkey_checked_at",
```
Comment (stebenz) [inline internal/query/sessions_test.go:103]: Same as above

File: internal/query/sessions_test.go [hunk 4]
```diff
@@ -102,6 +101,7 @@ var (
...
+		"u2f_checked_at",
```
Comment (stebenz) [inline internal/query/sessions_test.go:104]: Same as above

**Review by stebenz — CHANGES_REQUESTED** [review]

File: internal/command/session.go [hunk 5]
```diff
@@ -159,15 +156,23 @@ func (s *SessionCommands) IntentChecked(ctx context.Context, checkedAt time.Time
...
 }
 
-func (s *SessionCommands) PasskeyChecked(ctx context.Context, checkedAt time.Time, tokenID string, signCount uint32) {
+func (s *SessionCommands) WebAuthNChecked(ctx context.Context, checkedAt time.Time, tokenID string, signCount uint32, userVerified bool) {
 	s.eventCommands = append(s.eventCommands,
-		session.NewPasskeyCheckedEvent(ctx, s.sessionWriteModel.aggregate, checkedAt),
-		usr_repo.NewHumanPasswordlessSignCountChangedEvent(ctx, s.sessionWriteModel.aggregate, tokenID, signCount),
+		session.NewWebAuthNCheckedEvent(ctx, s.sessionWriteModel.aggregate, checkedAt, userVerified),
 	)
+	if s.sessionWriteModel.WebAuthNChallenge.UserVerification == domain.UserVerificationRequirementRequired {
```
[the remark this answers is not in the document]
Comment (muhlemmer) [reply internal/command/session.go:167]: That's how the "old" models are in zitadel. `passwordless` always requires User Verification. While `u2f` does not require this. In the new session API we express everything as a single `WebAuthN`, but untill the storage of `u2f` and `passwordless` are merged we should still distinguish between them.

This is practically what we (me and Livio) where discussing in the call yesterday.

File: internal/command/session_model.go [hunk 6]
```diff
@@ -170,8 +166,12 @@ func (wm *SessionWriteModel) AuthMethodTypes() []domain.UserAuthMethodType {
 	if !wm.PasswordCheckedAt.IsZero() {
 		types = append(types, domain.UserAuthMethodTypePassword)
 	}
-	if !wm.PasskeyCheckedAt.IsZero() {
-		types = append(types, domain.UserAuthMethodTypePasswordless)
+	if !wm.WebAuthNCheckedAt.IsZero() && wm.WebAuthNUserVerified {
+		if wm.WebAuthNUserVerified {
```
[the remark this answers is not in the document]
Comment (muhlemmer) [reply internal/command/session_model.go:170]: That was indeed an error. The `wm.WebAuthNUserVerified` on line 169 shouldn't have been there. I will remove it.

File: internal/query/sessions_test.go (hunk 1 above)
Comment (muhlemmer) [reply internal/query/sessions_test.go:82]: yes, you are right. Weird that the tests didn't fail. Appearantly you can put any arbirary string in the column names as long as the amount is correct...

**Review by stebenz — DISMISSED** [review]

**Review by stebenz — APPROVED** [review]

**Pull request merged**